projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -1852,7 +1852,7 @@ export abstract class IgxGridBaseDirective implements GridType,
     }
 
     public get resourceStrings(): IGridResourceStrings {
-        return this._resourceStrings;
+        return this._resourceStrings || this._defaultResourceStrings;
```

Comment:
Are we going to continue supporting overwriting just a subset of the strings via the property? Like in the docs: https://www.infragistics.com/products/ignite-ui-angular/angular/components/general/localization#localize-specific-strings-for-a-specific-instance-of-a-component

Because currently the way its setup, if you set a single string as custom, all others will resolve as undefined aka you have to overwrite all of the strings yourself.

Reply:
Yes, at least for now that is the goal. I've yet to mark them as `deprecated`, but I'll mark them and in a future release we can get rid of them.
Hm yeah the second thing sounds like an issue, I'll look into it. The rest resource strings should default to `en`

Reply:
If you set it initially via the property binding it breaks, for example:

```
<igx-grid [resourceStrings]="partialCustomHindi" ...
```

All other strings show as undefined:

<img width="1165" height="556" alt="image" src="https://github.com/user-attachments/assets/2cc1649e-92d2-494c-bbdb-b08f16828143" />

It does work if you first load a locale and then change it. But it does not, if you set it initially via the prop.